src/kbc_modules/cc_kbc/attester/tdx/mod.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::Attester;
+use anyhow::*;
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+use tdx_attest_rs;
+
+const CCEL_PATH: &str = "/sys/firmware/acpi/tables/data/CCEL";
+
+pub fn detect_platform() -> bool {
+    Path::new("/dev/tdx-attest").exists()
+}
+
+#[derive(Serialize, Deserialize)]
+struct TdxEvidence {
+    // Base64 encoded CC Eventlog ACPI table
+    // refer to https://uefi.org/specs/ACPI/6.5/05_ACPI_Software_Programming_Model.html#cc-event-log-acpi-table.
+    cc_eventlog: Option<String>,
+    // Base64 encoded TD quote.
+    quote: String,
+}
+
+#[derive(Debug, Default)]
+pub struct TdxAttester {}
+
+impl Attester for TdxAttester {
+    fn get_evidence(&self, report_data: String) -> Result<String> {
+        let mut report_data_bin = base64::decode(report_data.clone())?;
+        if report_data.len() != 48 {
+            return Err(anyhow!(
+                "TDX Attester: Report data should be SHA384 base64 String"
+            ));
+        }
+        report_data_bin.extend([0; 16]);
+
+        let tdx_report_data = tdx_attest_rs::tdx_report_data_t {
+            d: report_data_bin.as_slice().try_into()?,
+        };
+
+        let (get_quote_result, quote) =
+            tdx_attest_rs::tdx_att_get_quote(Some(&tdx_report_data), None, None, 0);
+        if get_quote_result != tdx_attest_rs::tdx_attest_error_t::TDX_ATTEST_SUCCESS
+            || quote.is_none()
+        {
+            return Err(anyhow!(
+                "TDX Attester: Failed to get TD quote. Error code: {:?}",
+                get_quote_result
+            ));
+        }
+
+        let cc_eventlog = match std::fs::read(CCEL_PATH) {
+            Result::Ok(el) => Some(base64::encode(el)),
+            Result::Err(_) => None,
```

Comment:
should we log (`warn!()`) in this case so we know something is not exactly right?

Reply:
Resolved.